pkg/api/types.go

```diff
@@ -799,6 +799,10 @@ type PodSpec struct {
 	// Required: there must be at least one container in a pod.
 	Containers    []Container   `json:"containers"`
 	RestartPolicy RestartPolicy `json:"restartPolicy,omitempty"`
+	// Optional duration in seconds the pod needs to terminate gracefully. May be decreased in delete request.
+	// Value must be non-negative integer. The value zero indicates delete immediately.
+	// If this value is nil, the default grace period will be used instead.
+	TerminationGracePeriodSeconds *int64 `json:"terminationGracePeriodSeconds,omitempty" description:"duration the pod needs to terminate gracefully"`
```

Comment:
As part of this Godoc and description, just add the extra sentences.  Want to be users can draw the link between this value and the sigterm/sigkill behavior of their pods (although we don't have to say sigkill sigterm specifically right now which is why I said termination and kill signals).